ca/ca.go

```diff
@@ -433,16 +433,21 @@ func (ca *certificateAuthorityImpl) issuePrecertificateInner(ctx context.Context
 	ca.log.AuditInfof("Signing precert: serial=[%s] regID=[%d] names=[%s] csr=[%s]",
 		serialHex, issueReq.RegistrationID, strings.Join(csr.DNSNames, ", "), hex.EncodeToString(csr.Raw))
 
-	certDER, err := issuer.Issue(&issuance.IssuanceRequest{
+	names := csrlib.NamesFromCSR(csr)
+	req := &issuance.IssuanceRequest{
 		PublicKey:         csr.PublicKey,
 		Serial:            serialBigInt.Bytes(),
-		CommonName:        csr.Subject.CommonName,
-		DNSNames:          csr.DNSNames,
+		DNSNames:          names.SANs,
 		IncludeCTPoison:   true,
 		IncludeMustStaple: issuance.ContainsMustStaple(csr.Extensions),
 		NotBefore:         validity.NotBefore,
 		NotAfter:          validity.NotAfter,
-	})
+	}
+	if !features.Enabled(features.OmitCommonName) {
```

Comment:
Note: the `features` package supports feature flags that default to true, exactly so we can avoid this sort of double negative. If you add a feature flag `SetCommonName`, defaulting to true, this would read as:

```suggestion
	if features.Enabled(features.SetCommonName) {
```